indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -629,6 +648,207 @@ public Response getCompleteTasks(
     return Response.ok(completeTasks).build();
   }
 
+  @GET
+  @Path("/tasks")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getTasks(
+      @QueryParam("state") final String state,
+      @QueryParam("datasource") final String dataSource,
+      @PathParam("interval") final String interval,
+      @QueryParam("maxCompletedTasks") final Integer maxCompletedTasks,
+      @Context final HttpServletRequest req
+  )
+  {
+
+    if (state != null) {
+      switch (StringUtils.toUpperCase(state)) {
+        case "PENDING":
+          return workItemsResponse(
```

Comment:
There are already endpoints to get pending, waiting, running, or complete tasks. Please share the code path rather than duplicating.

Reply:
Yes, the idea is to potentially call this `tasks` api with appropriate `state` query param from the existing endpoints to get pending, waiting, running, complete tasks. I did not do that change yet as this api is still being tested in nightly, will do that once this api passes those tests.